src/time.rs

```diff
@@ -0,0 +1,149 @@
+use sys::time::TimeSpec;
+use std::mem::MaybeUninit;
+use {Result, Errno};
+use libc;
+
+libc_enum! {
+    #[cfg_attr(any(
+                target_os = "fuchsia",
+                target_os = "redox",
+                target_env = "uclibc",
+                target_os = "freebsd",
+                target_os = "netbsd",
+                target_os = "openbsd",
+                target_os = "haiku",
+                target_os = "linux",
+                target_os = "android",
+                target_os = "emscripten",
+                target_os = "solaris",
+                target_os = "illumos",
+                all(target_env = "newlib", target_arch = "arm"),
```

Comment:
Why the special case for `target_arch` on newlib?  I don't see that in libc.

Reply:
For `newlib`, the supported `target_archs` are only `arm `and `aarch64`. While the type `clockid_t` is defined here:
https://github.com/rust-lang/libc/blob/49c1d138e3b4719e21ee7cbce7dd903543cda80d/src/unix/newlib/mod.rs#L4

`c_ulong` is defined for each arch in their respective modules: 
https://github.com/rust-lang/libc/blob/49c1d138e3b4719e21ee7cbce7dd903543cda80d/src/unix/newlib/arm/mod.rs#L5
https://github.com/rust-lang/libc/blob/49c1d138e3b4719e21ee7cbce7dd903543cda80d/src/unix/newlib/aarch64/mod.rs#L5

Reply:
Just realized that I had put the "arm" case under i32 rather than u32. Hopefully makes more sense now.